src/composables/latestTransactionList.ts

```diff
@@ -1,62 +1,214 @@
+      ));
+      areLatestTransactionsUpdating.value = false;
+    }
+  }
+
+  function removeAccountPendingTransaction(address: AccountAddress, hash: string) {
+    accountsTransactionsPending.value[address] = remove(
+      accountsTransactionsPending.value[address],
+      (transaction) => hash === transaction.hash,
+    );
+  }
+
+  /**
+   * Add temporary pending transaction and remove it when it's mined.
+   * For now this is usable only with Aeternity protocol
+   */
+  async function addAccountPendingTransaction(address: AccountAddress, transaction: ITransaction) {
+    const account = getAccountByAddress(address);
+    if (account?.protocol === PROTOCOLS.aeternity && transaction?.hash) {
+      if (!accountsTransactionsPending.value[address]) {
+        accountsTransactionsPending.value[address] = [];
+      }
+      accountsTransactionsPending.value[address].push(transaction);
+
+      try {
+        await waitTransactionMined(transaction.hash);
+        loadAccountLatestTransactions(account);
+      } finally {
+        removeAccountPendingTransaction(address, transaction.hash);
+      }
+    }
+  }
+
+  if (!initialized) {
+    initialized = true;
+    loadAllLatestTransactions();
+
+    /**
+     * Remove old pending transactions that we can consider as already mined.
+     * This prevents situation where user creates transaction and closes the app/extension
+     * immediately so the `waitTransactionMined` couldn't work properly.
+     */
+    setInterval(() => {
+      Object.entries(accountsTransactionsPending.value)
+        .forEach(([accountAddress, transactionList]) => {
+          transactionList.forEach(({ hash, microTime }) => {
+            if (Date.now() - (microTime || 0) > TRANSACTION_CERTAINLY_MINED_TIME) {
+              removeAccountPendingTransaction(accountAddress, hash);
+            }
           });
-          return {
-            ...tr,
-            direction: direction.value,
-          };
         });
-      })
-      .flatMap((transaction) => transaction);
+    }, 60000);
+
+    /**
+     * To avoid unnecessary data transfers, instead of polling data at intervals,
+     * we only load account transactions when the account balance changes.
+     */
+    watch(
+      balances,
+      (newBalances, oldBalances) => {
+        accounts.value.forEach((account) => {
+          const oldBalance = oldBalances[account.address];
+          const newBalance = newBalances[account.address];
+          if (oldBalance && newBalance && !newBalance.isEqualTo(oldBalance)) {
+            setTimeout(
+              () => loadAccountLatestTransactions(account),
+              (account.protocol === PROTOCOLS.aeternity) ? AE_MDW_TO_NODE_APPROX_DELAY_TIME : 0,
+            );
+          }
+        });
+      },
+      { deep: true },
+    );
+
+    watch(
+      tokenBalances,
+      (oldTokens, newTokens) => {
+        if (!isEqual(oldTokens, newTokens)) {
+          loadAllLatestTransactions();
```

Comment:
Shall we also split the logic to load only one protocol specific latest transactions, in case only token balance of one protocol changed? Something to consider for later improvements of course.